### PR TITLE
Remove operator logic and use subctl for deployment

### PR DESCRIPTION
**Pull request by mangelajo**
All operator verification logic has been removed, as those verifications
are already handled by the submariner-operator repository.

The deploynent has been switched to subctl, using the latest binary
release.

### CODE REVIEW

**Comment by mangelajo** [comment]
This change is [Reviewable](https://reviewable.io/reviews/submariner-io/submariner/231)

**Comment by mangelajo** [comment]
@dimaunx @dfarrell07 @skitt 
I suggest we separate the helm/subctl logic to completely different files, may be with

$x_setup_broker cluster1
$x_setup_connectivity cluster2
$x_setup_connectivity cluster2

and those are implemented with

helm_setup_connectivity
helm_setup_broker

operator_setup_connectivity
operator_setup_broker

etc... and any common variables we just pass them as env vars (arrays or just vars...)

That way we separate all the helm/operator logic now a little bit mixed up.

**Comment by skitt** [comment]
> @dimaunx @dfarrell07 @skitt
> I suggest we separate the helm/subctl logic to completely different files, may be with
> 
> $x_setup_broker cluster1
> $x_setup_connectivity cluster2
> $x_setup_connectivity cluster2
> 
> and those are implemented with
> 
> helm_setup_connectivity
> helm_setup_broker
> 
> operator_setup_connectivity
> operator_setup_broker
> 
> etc... and any common variables we just pass them as env vars (arrays or just vars...)
> 
> That way we separate all the helm/operator logic now a little bit mixed up.

That would make sense in the long run, but I’d like to see @dimaunx’s plans for e2e unification before we spend too much time on refactoring all this...

**Comment by mangelajo** [comment]
I have verified this works also gracefully with helm:

```
Ran 11 of 11 Specs in 154.587 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestE2E (154.59s)
PASS
ok      github.com/submariner-io/submariner/test/e2e    154.607s
your 3 virtual clusters are deployed and working properly with your local
submariner source code, and can be accessed with:

export KUBECONFIG=$(echo $(git rev-parse --show-toplevel)/output/kind-config/local-dev/kind-config-cluster{1..3} | sed 's/ /:/g')

$ kubectl config use-context cluster1 # or cluster2, cluster3..

to cleanup, just run: make e2e status=clean
[submariner]$ ./scripts/e2e keep 1.14.2 false false helm false
```

File: scripts/kind-e2e/e2e.sh [hunk 1]
```diff
@@ -123,6 +125,23 @@ function setup_broker() {
     SUBMARINER_BROKER_TOKEN=$(kubectl --context=cluster1 -n ${SUBMARINER_BROKER_NS} get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${SUBMARINER_BROKER_NS}-client')].data.token}"|base64 --decode)
 }
 
+function setup_broker_subctl() {
+    if kubectl --context=cluster1 get crd clusters.submariner.io > /dev/null 2>&1; then
+        echo Submariner CRDs already exist, skipping broker creation...
+    else
+        echo Installing broker on cluster1.
+        subctl --kubeconfig ${PRJ_ROOT}/output/kind-config/dapper/kind-config-cluster1 deploy-broker --no-dataplane
```
Comment (manosnoam) [inline scripts/kind-e2e/e2e.sh]: Will it work with the fix of #75 (Flip the dataplane subctl flag) ?

Comment (mangelajo) [reply scripts/kind-e2e/e2e.sh]: We will update that flag when we update to subctl 0.0.3 :-), 0.0.2 should remain the same

File: scripts/kind-e2e/e2e.sh [hunk 2]
```diff
@@ -382,97 +394,48 @@ if [[ $4 = true ]]; then
...
-      subm_ns=submariner
-
-      verify_subm_engine_deployment
-      verify_subm_engine_pod
-      verify_subm_routeagent_daemonset
-      verify_subm_routeagent_pod
-      verify_subm_engine_container
-      verify_subm_routeagent_container
-      verify_subm_engine_secrets
-      verify_subm_routeagent_secrets
```
Comment (manosnoam) [inline scripts/kind-e2e/e2e.sh]: Why did you remove the verifying functions, shouldn't we check that subctl did everything ?
